### PR TITLE
Remove deprecated M-API methods

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

**Breaking changes:**

- Removed deprecated magic `__call()` method on `ApiClient` class
- Removed deprecated `ClientGrants::get()` method
- Removed deprecated `Users:: search()` method
- Removed deprecated `Users:: unlinkDevice()` method

### Checklist

- [x] All existing and new tests complete without errors.
- [x] The correct base branch is being used.

### CODE REVIEW

File: src/API/Helpers/ApiClient.php [hunk 1]
```diff
@@ -22,7 +22,10 @@ public static function setInfoHeadersData(InformationHeaders $infoHeadersData)
         self::$infoHeadersData = $infoHeadersData;
     }
 
-    public static function getInfoHeadersData()
+    /**
+     * @return InformationHeaders|null
+     */
+    public static function getInfoHeadersData() : ?InformationHeaders
```
Comment (damieng) [inline src/API/Helpers/ApiClient.php]: Worth adding some doc comments about the function itself as well?

File: src/API/Helpers/ApiClient.php [hunk 2]
```diff
@@ -55,38 +58,15 @@ public function __construct($config)
     {
         $this->basePath      = $config['basePath'];
         $this->domain        = $config['domain'];
-        $this->returnType    = isset( $config['returnType'] ) ? $config['returnType'] : null;
-        $this->headers       = isset($config['headers']) ? $config['headers'] : [];
-        $this->guzzleOptions = isset($config['guzzleOptions']) ? $config['guzzleOptions'] : [];
+        $this->returnType    = $config['returnType'] ?? null;
```
Comment (damieng) [inline src/API/Helpers/ApiClient.php]: ?? is null coalescing... so isn't this a no-op?

Comment (joshcanhelp) [reply src/API/Helpers/ApiClient.php]: You can't just use `$config['returnType']` or you get an error:

```
❯ php -r '$arr = []; echo $arr["nothing"];'
PHP Notice:  Undefined index: nothing in Command line code on line 1
PHP Stack trace:
PHP   1. {main}() Command line code:0

Notice: Undefined index: nothing in Command line code on line 1

Call Stack:
    0.0002     359800   1. {main}() Command line code:0

❯ php -r '$arr = []; echo $arr["nothing"] ?? "👍";'
👍
```

File: src/API/Helpers/ApiClient.php (hunk 1 above)
Comment (joshcanhelp) [reply src/API/Helpers/ApiClient.php]: I mean ... yeah, probably 😆

File: src/API/Helpers/ApiClient.php (hunk 2 above)
Comment (damieng) [reply src/API/Helpers/ApiClient.php]: Ah so '??' is null AND nothing coalescing in php?

Comment (joshcanhelp) [reply src/API/Helpers/ApiClient.php]: More or less :)

**Pull request merged**